runtime/compiler/env/J9SharedCache.cpp

```diff
@@ -813,76 +829,155 @@ TR_J9SharedCache::findChainForClass(J9Class *clazz, const char *key, uint32_t ke
    return chainForClass;
    }
 
+bool
+TR_J9SharedCache::validateSuperClassesInClassChain(J9Class *clazz, UDATA * & chainPtr, UDATA *chainEnd)
+   {
+   int32_t numSuperclasses = TR::Compiler->cls.classDepthOf(fe()->convertClassPtrToClassOffset(clazz));
+   for (int32_t index=0; index < numSuperclasses;index++)
+      {
+      J9ROMClass *romClass = TR::Compiler->cls.romClassOfSuperClass(fe()->convertClassPtrToClassOffset(clazz), index);
```

Comment:
Let's cache `fe()->convertClassPtrToClassOffset(clazz)` and take it out of the loop